swaylock/render.c

```diff
@@ -18,7 +18,15 @@ static void set_color_for_state(cairo_t *cairo, struct swaylock_state *state,
 	} else if (state->auth_state == AUTH_STATE_CLEAR) {
 		cairo_set_source_u32(cairo, colorset->cleared);
 	} else {
-		cairo_set_source_u32(cairo, colorset->input);
+		if (state->xkb.caps_lock && state->args.show_caps_lock_indicator) {
+			cairo_set_source_u32(cairo, colorset->caps_lock);
+		} else if (state->xkb.caps_lock && !state->args.show_caps_lock_indicator) {
```

Comment:
Can you also add a check for show_caps_lock_text here since both can be disabled and some other form of text may be shown?

Tbh I'm not a huge fan of setting the font color here, but moving it would require changes that don't belong in this PR

Reply:
I'm not a fan of this either, but unfortunatly there was no "easy" way of setting just one color.
Nonetheless I've added the check